soc/xtensa/esp32/linker.ld

```diff
@@ -276,8 +296,88 @@ SECTIONS
     *(.gnu.version_d)
     . = ALIGN(4);               /* this table MUST be 4-byte aligned */
     _rodata_end = ABSOLUTE(.);
+  } GROUP_LINK_IN(ROMABLE_REGION)
+
+#pragma push_macro("Z_ITERABLE_SECTION_ROM")
+#pragma push_macro("ROMABLE_REGION")
+#undef Z_ITERABLE_SECTION_ROM
+#define Z_ITERABLE_SECTION_ROM(x,y)
+#undef ROMABLE_REGION
+/* This is to workaround limitation of `esptool` which needs single `FLASH` data segment
+ * which is already defined above. In case, `common-rom.ld` creates additional segments
+ * they will be placed in DRAM instead. */
+#define ROMABLE_REGION RAMABLE_REGION
+#include <linker/common-rom.ld>
+/* Restore original value for symbols referenced by `common-rom.ld` */
+__log_const_start = __esp_log_const_start;
+__log_const_end = __esp_log_const_end;
+__log_backends_start = __esp_log_backends_start;
```

Comment:
Oh, they're coming back here.  This seems a little complex.  I guess it's necessary.